src/main/java/com/timgroup/statsd/NonBlockingStatsDClient.java

```diff
@@ -1093,19 +1108,20 @@ StringBuilder tagString(final String[] tags, StringBuilder builder) {
         return tagString(tags, constantTagsRendered, builder);
     }
 
-    abstract class StatsDMessage implements Message {
-        final String aspect;
-        final String type;
+    abstract class StatsDMessage<T extends Number> extends Message<T> {
         final double sampleRate; // NaN for none
         final String[] tags;
 
-        protected StatsDMessage(String aspect, String type, double sampleRate, String[] tags) {
-            this.aspect = aspect;
-            this.type = type;
+        protected StatsDMessage(String aspect, Message.Type type, T value, double sampleRate, String[] tags) {
+            super(aspect, type, value);
```

Comment:
The tags should be included in the `aspect`, no ? Since we're using it to hash metrics in the aggregator.